x/yarpctest/types/handler_unary.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package types
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/atomic"
+	"go.uber.org/multierr"
+	"go.uber.org/yarpc/api/middleware"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/internal/inboundmiddleware"
+	"go.uber.org/yarpc/x/yarpctest/api"
+)
+
+// UnaryHandler is a struct that implements the ProcOptions and HandlerOption
+// interfaces (so it can be used directly as a procedure, or as a single use
+// handler (depending on the use case)).
+type UnaryHandler struct {
+	H  api.UnaryHandler
+	MW []api.UnaryInboundMiddleware
```

Comment:
Tempting, but let’s not abbreviate. Fine to truncate to `Handler` and `Middleware` since they’re unambiguous in the context of `UnaryHandler`.